src/target/stm32g0.c

```diff
@@ -0,0 +1,758 @@
+#define FLASH_CR_PG (1U << 0U)
+
+#define FLASH_SR (G0_FLASH_BASE + 0x010)
+#define FLASH_SR_CFGBSY (1U << 18U)
+#define FLASH_SR_BSY2 (1U << 17U)
+#define FLASH_SR_BSY1 (1U << 16U)
+#define FLASH_SR_OPTVERR (1U << 15U)
+#define FLASH_SR_RDERR (1U << 14U)
+#define FLASH_SR_FASTERR (1U << 9U)
+#define FLASH_SR_MISSERR (1U << 8U)
+#define FLASH_SR_PGSERR (1U << 7U)
+#define FLASH_SR_SIZERR (1U << 6U)
+#define FLASH_SR_PGAERR (1U << 5U)
+#define FLASH_SR_WRPERR (1U << 4U)
+#define FLASH_SR_PROGERR (1U << 3U)
+#define FLASH_SR_OPERR (1U << 1U)
+#define FLASH_SR_EOP (1U << 0U)
+#define FLASH_SR_ERROR_MASK                                                    \
+	(FLASH_SR_OPTVERR | FLASH_SR_RDERR | FLASH_SR_FASTERR |                \
+	 FLASH_SR_MISSERR | FLASH_SR_PGSERR | FLASH_SR_SIZERR |                \
+	 FLASH_SR_PGAERR | FLASH_SR_WRPERR | FLASH_SR_PROGERR |                \
+	 FLASH_SR_OPERR)
+#define FLASH_SR_BSY_MASK (FLASH_SR_BSY2 | FLASH_SR_BSY1)
+
+#define FLASH_OPTKEYR (G0_FLASH_BASE + 0x00C)
+#define FLASH_OPTKEYR_KEY1 0x08192A3B
+#define FLASH_OPTKEYR_KEY2 0x4C5D6E7F
+#define FLASH_OPTR (G0_FLASH_BASE + 0x020)
+#define FLASH_OPTR_DUAL_BANK (1U << 21U)
+#define FLASH_OPTR_RDP_MASK 0xFF
+#define FLASH_PCROP1ASR (G0_FLASH_BASE + 0x024)
+#define FLASH_PCROP1AER (G0_FLASH_BASE + 0x028)
+#define FLASH_PCROP1AER_PCROP_RDP (1U << 31U)
+#define FLASH_WRP1AR (G0_FLASH_BASE + 0x02C)
+#define FLASH_WRP1BR (G0_FLASH_BASE + 0x030)
+#define FLASH_PCROP1BSR (G0_FLASH_BASE + 0x034)
+#define FLASH_PCROP1BER (G0_FLASH_BASE + 0x038)
+#define FLASH_PCROP2ASR (G0_FLASH_BASE + 0x044)
+#define FLASH_PCROP2AER (G0_FLASH_BASE + 0x048)
+#define FLASH_WRP2AR (G0_FLASH_BASE + 0x04C)
+#define FLASH_WRP2BR (G0_FLASH_BASE + 0x050)
+#define FLASH_PCROP2BSR (G0_FLASH_BASE + 0x054)
+#define FLASH_PCROP2BER (G0_FLASH_BASE + 0x058)
+#define FLASH_SECR (G0_FLASH_BASE + 0x080)
+
+/* RAM */
+#define RAM_START 0x20000000
+#define RAM_SIZE_G03_4 (8U * 1024U) // 8 kB
+#define RAM_SIZE_G05_6 (18U * 1024U) // 18 kB
+#define RAM_SIZE_G07_8 (36U * 1024U) // 36 kB
+#define RAM_SIZE_G0B_C (144U * 1024U) // 144 kB
+
+/* RCC */
+#define G0_RCC_BASE 0x40021000
+#define RCC_APBENR1 (G0_RCC_BASE + 0x3C)
+#define RCC_APBENR1_DBGEN (1U << 27U)
+
+/* DBGMCU */
+#define DBG_BASE 0x40015800
+#define DBG_IDCODE (DBG_BASE + 0x00)
+#define DBG_DEV_ID_MASK 0x00000FFF
+#define DBG_REV_ID_MASK 0xFFFF0000
+#define DBG_CR (DBG_BASE + 0x04)
+#define DBG_CR_DBG_STANDBY (1U << 2U)
+#define DBG_CR_DBG_STOP (1U << 1U)
+#define DBG_APB_FZ1 (DBG_BASE + 0x08)
+#define DBG_APB_FZ2 (DBG_BASE + 0x0C)
+#define DBG_APB_FZ1_DBG_IWDG_STOP (1U << 12U)
+#define DBG_APB_FZ1_DBG_WWDG_STOP (1U << 11U)
+
+enum STM32G0_DEV_ID {
+	STM32G03_4 = 0x466,
+	STM32G05_6 = 0x456,
+	STM32G07_8 = 0x460,
+	STM32G0B_C = 0x467
+};
+struct saved_regs_s {
+	uint32_t rcc_apbenr1;
+	uint32_t dbg_cr;
+	uint32_t dbg_apb_fz1;
+};
+
+static struct saved_regs_s saved_regs = {0};
+static bool irreversible_enabled = false;
```

Comment:
Why not as additional entity in saved_regs

Reply:
Yes. I am waiting for knowing how to properly store "saved_regs".